src/cmd/build.js

```diff
@@ -103,15 +111,49 @@ async function defaultPackageCreator({
   let packageName = safeFileName(
     `${extensionName}-${manifestData.version}.zip`);
   let extensionPath = path.join(artifactsDir, packageName);
-  let stream = createWriteStream(extensionPath);
 
-  stream.write(buffer, () => stream.end());
+  let destinationFileExists = await fs.stat(extensionPath)
+    .then((stat)=>{
+      return stat.isFile();
+    })
+    .catch(()=>{
+      return false;
+    })
+  ;
 
-  await streamToPromise(stream);
+  function throwExtensionExists() {
+    throw new UsageError(
+        `Extension exists at the destination path: ${extensionPath}\n` +
+        'Use --overwrite-dest to enable overwriting.');
+  }
 
-  if (showReadyMessage) {
-    log.info(`Your web extension is ready: ${extensionPath}`);
+  if (destinationFileExists) {
+    if (overwriteDest) {
+      log.info(`Destination exists, overwriting: ${extensionPath}`);
+    } else {
+      throwExtensionExists();
+    }
   }
+
+  let stream = overwriteDest ?
+	createWriteStream(extensionPath)
+	: createWriteStream(extensionPath, {flags: 'wx'}); // double-check if destination file exists
```

Comment:
can't you always just make this `createWriteStream(extensionPath, {flags: 'wx'})` regardless of whether we need to overwrite the destination or not?